python-package/lightgbm/plotting.py

```diff
@@ -414,13 +415,38 @@ def plot_metric(
     return ax
 
 
+def _determine_direction_for_numeric_split(fval: float, threshold: float, missing_type: str, default_left: bool) -> str:
+    le_threshold = fval <= threshold
+    if missing_type == _MissingType.NONE or (missing_type == _MissingType.ZERO and default_left and ZERO_THRESHOLD < threshold):
```

Comment:
You cannot compare string with Enum this way. Such comparison will always return `False.`
```python
missing_type = 'Zero'
missing_type == _MissingType.ZERO  # False
```
`missing_type` should either be converted to Enum first or you should compare in the following way: `missing_type == _MissingType.ZERO.value`.

Also, this incorrect comparison shows that added tests are unreliable unfortunately 😢 They should fail, but all CI is green right now.

Reply:
I updated the numerical split definition to the one here: https://github.com/microsoft/LightGBM/blob/f94050a4cc94909e10a0064baff11cec795eb250/include/LightGBM/tree.h#L335-L353.
Now the test fails if I comment out the enum conversion. Let me know what you think, it looks different than the previous one.

Reply:
Thanks for doing this!

Do we need to sync the code in `if/else` dump as well (in a separate PR)?
https://github.com/microsoft/LightGBM/blob/11110c540eb84f519da299ea9eb76cc23a0ca570/src/io/tree.cpp#L520-L560

Reply:
Yes, I think that code may need to be updated. I'll take a look.

Reply:
Kindly ping @jmoralez for possible following-up PR.